google/resource_compute_disk.go

```diff
@@ -390,213 +747,359 @@ func resourceComputeDiskDelete(d *schema.ResourceData, meta interface{}) error {
+		req = append(req, outMap)
+	} else {
+		// Check alternative setting?
+		if altV, ok := d.GetOk("disk_encryption_key_raw"); ok {
+			outMap := make(map[string]interface{})
+			outMap["rawKey"] = altV
+			req = append(req, outMap)
+		}
 	}
+	return req, nil
+}
 
-	if suppressWindowsFamilyDiff(imageName, familyName) {
-		return true
+func expandComputeDiskSourceImageEncryptionKey(v interface{}, d *schema.ResourceData, config *Config) (interface{}, error) {
+	l := v.([]interface{})
+	req := make([]interface{}, 0, 1)
+	if len(l) == 1 {
+		// There is a value
+		outMap := make(map[string]interface{})
+		outMap["rawKey"] = l[0].(map[string]interface{})["raw_key"]
+		req = append(req, outMap)
 	}
+	return req, nil
+}
 
-	return false
+func expandComputeDiskSnapshot(v interface{}, d *schema.ResourceData, config *Config) (interface{}, error) {
+	return v, nil
 }
 
-// e.g. image: ubuntu-1404-trusty-v20180122, family: ubuntu-1404-lts
-func suppressCanonicalFamilyDiff(imageName, familyName string) bool {
-	parts := canonicalUbuntuLtsImage.FindStringSubmatch(imageName)
-	if len(parts) == 2 {
-		f := fmt.Sprintf("ubuntu-%s-lts", parts[1])
-		if f == familyName {
-			return true
-		}
+func expandComputeDiskSourceSnapshotEncryptionKey(v interface{}, d *schema.ResourceData, config *Config) (interface{}, error) {
+	l := v.([]interface{})
+	req := make([]interface{}, 0, 1)
+	if len(l) == 1 {
+		// There is a value
+		outMap := make(map[string]interface{})
+		outMap["rawKey"] = l[0].(map[string]interface{})["raw_key"]
+		req = append(req, outMap)
 	}
-
-	return false
+	return req, nil
 }
 
-// e.g. image: sql-2017-standard-windows-2016-dc-v20180109, family: sql-std-2017-win-2016
-// e.g. image: sql-2017-express-windows-2012-r2-dc-v20180109, family: sql-exp-2017-win-2012-r2
-func suppressWindowsSqlFamilyDiff(imageName, familyName string) bool {
-	parts := windowsSqlImage.FindStringSubmatch(imageName)
-	if len(parts) == 5 {
-		edition := parts[2] // enterprise, standard or web.
-		sqlVersion := parts[1]
-		windowsVersion := parts[3]
+func resourceComputeDiskEncoder(d *schema.ResourceData, meta interface{}, obj map[string]interface{}) (map[string]interface{}, error) {
+	config := meta.(*Config)
 
-		// Translate edition
-		switch edition {
-		case "enterprise":
-			edition = "ent"
-		case "standard":
-			edition = "std"
-		case "express":
-			edition = "exp"
-		}
+	project, err := getProject(d, config)
+	if err != nil {
+		return nil, err
+	}
 
-		var f string
-		if revision := parts[4]; revision != "" {
-			// With revision
-			f = fmt.Sprintf("sql-%s-%s-win-%s-r%s", edition, sqlVersion, windowsVersion, revision)
-		} else {
-			// No revision
-			f = fmt.Sprintf("sql-%s-%s-win-%s", edition, sqlVersion, windowsVersion)
+	// Get the zone
+	z, err := getZone(d, config)
+	if err != nil {
+		return nil, err
+	}
+
+	zone, err := config.clientCompute.Zones.Get(project, z).Do()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
I think you need to add something here too for snapshots

Reply:
You're right!  Dang!  How did you know that?